# Data visualization
- Intro to Data Science
- Karlo Boroš - 0036531517
- Dataset: "Pima Indians diabetes mellitus classification based on machine learning (ML) algorithms"

### Vizualizacija podataka
- Uvod u znanost o podacima
- Karlo Boroš - 0036531517
- Zadani skup podataka: "Pima Indians diabetes mellitus classification based on machine learning (ML) algorithms"

### 0. Učitavanje potrebnih paketa

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

### 1. Priprema podataka

#### a) Učitavanje skupa i prvi pogled na podatke

In [2]:
data = pd.read_csv("diabetes.csv")

In [3]:
data.shape

(768, 9)

In [41]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [4]:
data.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [5]:
data.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [6]:
data.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                51
Insulin                     186
BMI                         248
DiabetesPedigreeFunction    517
Age                          52
Outcome                       2
dtype: int64

In [7]:
data.isna().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

Iz prethodnih linija koda i njihovih izlaza možemo izvesti nekoliko osnovnih zaključaka:
- dataset se sastoji od 768 primjera i 9 značajki koje razmatramo
- jedino je značajka Outcome binarna što je i logično jer nam govori ima li osoba dijabetes ili ne, dok ostale značajke poprimaju razne vrijednosti
- nemamo nijedan NA podatak što je odlično, ali očito su ti podaci zamijenjeni nulama što nam ne odgovara jer ne smiju biti nula tako da ću te podatke isključiti iz analiza ovisno o potrebi

U sljedećim redovima su objašnjenja pojedinih značajki sa stranice:
- Pregnancies - number of times pregnant
- Glucose - plasma glucose concentration a 2 hours in an oral glucose tolerance test
- BloodPressure - diastolic blood pressure (mm Hg)
- SkinThickness - triceps skin fold thickness (mm)
- Insulin - 2-Hour serum insulin (mu U/ml)
- BMI - body mass index (weight in kg/(height in m)^2)
- DiabetesPedigree - diabetes pedigree function
- Age - age (years)
- Outcome - class variable (0 or 1) 268 of 768 are 1, the others are 0

Moj cilj je analizirati značajke i probati doći do kvalitetnih zaključaka o predispozicijama za dijabetes. 

#### b) Prilagodba podataka za analizu

In [8]:
data_glucose = data[data.Glucose > 0]
data_blood = data[data.BloodPressure > 0]
data_skin = data[data.SkinThickness > 0]
data_insulin = data[data.Insulin > 0]
data_bmi = data[data.BMI > 0]

#print(min(data_glucose.Glucose))
#print(min(data_blood.BloodPressure))
#print(min(data_skin.SkinThickness))
#print(min(data_insulin.Insulin))
#print(min(data_bmi.BMI))

### 2. Vizualizacija podataka

#### y) X tip vizualizacije

#### y) X tip vizualizacije

#### y) X tip vizualizacije

#### y) X tip vizualizacije

#### y) X tip vizualizacije